### PR TITLE
Fix #6 : Throw a DuplicateTorrentException when Transmission returns this error

### LINKED ISSUE #6: Undefined index: torrent-added in RpcClient::torrentAdd for duplicates
This function returns `$response['arguments']['torrent-added']`.

Response may not contain `torrent-added` key, but contain `torrent-duplicate` key.

### PR DESCRIPTION
**Pull request by MartialGeek**

### CODE REVIEW

**Comment by popstas** [comment]
All right

File: src/RpcClient.php [hunk 1]
```diff
@@ -215,10 +215,24 @@ public function torrentAdd($sessionId, array $argumentsWithValues)
...
-            'torrent-add',
-            $argumentsWithValues
-        ));
+        try {
+            $response = $this->sendRequest($sessionId, $this->buildRequestBody(
+                'torrent-add',
+                $argumentsWithValues
+            ));
+        } catch (TransmissionException $e) {
+            if ($e->getResult() === 'duplicate torrent') {
```
Comment (popstas) [inline src/RpcClient.php]: Hello, I checked new release, TransmissionException not fires on duplicate torrent

File: src/RpcClient.php [hunk 2]
```diff
@@ -215,10 +215,24 @@ public function torrentAdd($sessionId, array $argumentsWithValues)
...
+                $duplicateTorrentException->setTorrentName($responseArguments['torrent-duplicate']['name']);
+                $duplicateTorrentException->setTorrentHashString($responseArguments['torrent-duplicate']['hashString']);
+
+                throw $duplicateTorrentException;
+            }
+
+            throw $e;
+        }
 
         return $response['arguments']['torrent-added'];
```
Comment (popstas) [inline src/RpcClient.php]: Sample data while trying add duplicate torrent:

```
Array
(
    [arguments] => Array
        (
            [torrent-duplicate] => Array
                (
                    [hashString] => 445d1c15212a0213baca29a4854f9dafa5bad863
                    [id] => 96
                    [name] => movie.mkv
                )
        )
    [result] => success
)
```

**Comment by MartialGeek** [comment]
OK,  maybe Transmission does not return the string 'duplicate torrent' in the result key?
I will take a look.

**Comment by MartialGeek** [comment]
Actually, the response in case of duplicate torrent is:
{"arguments":{"torrent-duplicate":{"hashString":"c388a78f357c70232ae3007d506bb6bc5cc43505","id":127,"name":"debian-8.3.0-amd64-CD-1.iso"}},"result":"success"}
I will fix that today.

**Comment by MartialGeek** [comment]
This commit should fix the problem:
https://github.com/MartialGeek/transmission-api/commit/9fdc30bbaaf485c8bc980ce37d40ef76c8912532

**Comment by popstas** [comment]
Yes, it works!

**Pull request merged**